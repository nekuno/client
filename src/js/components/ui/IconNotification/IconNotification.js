import PropTypes from 'prop-types';
import React, { Component } from 'react';
import styles from './IconNotification.scss';

export default class IconNotification extends Component {

    static propTypes = {
        icon           : PropTypes.string.isRequired,
        notifications  : PropTypes.number,
        onClickHandler : PropTypes.func.isRequired
    };

    handleClick() {
        this.props.onClickHandler();
    }

    render() {
        const {icon, notifications} = this.props;

        return (
            <div className={styles.iconNotification} onClick={this.handleClick.bind(this)}>
                <div className={styles.icon + ' icon icon-' + icon} />
                <div className={styles.notificationsWrapper}>
                    <div className={styles.notifications}>
                        {notifications > 99 ? 99 : notifications}
                    </div>
                </div>
            </div>
        );
    }
}