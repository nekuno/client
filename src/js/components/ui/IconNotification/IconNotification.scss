@import '../../../../scss/_partials/colors';

.icon-notification {
  cursor: pointer;
  padding: 5px 0 0 0;
  position: relative;
  height: 40px;
  float: right;
  width: 45px;

  .icon {
    color: #6C6F82;
    line-height: 50px;
    text-align: center;
    height: 50px;
    width: 40px;

    &::before {
      font-size: 25px;
    }
  }
  .notifications-wrapper {
    background: $deep-purple;
    border-radius: 50%;
    -moz-border-radius: 50%;
    -webkit-border-radius: 50%;
    position: absolute;
    text-align: center;
    line-height: 20px;
    top: 10px;
    right: 0;
    height: 20px;
    width: 20px;

    .notifications {
      color: #fff;
      line-height: 20px;
      font-size: 10px;
      font-weight: bold;
    }
  }
}