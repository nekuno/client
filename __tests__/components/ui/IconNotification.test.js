import React from 'react';
import IconNotification from '../../../src/js/components/ui/IconNotification/IconNotification.js';
import styles from '../../../src/js/components/ui/IconNotification/IconNotification.scss';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Test IconNotification component', () => {
    const mockFn = jest.fn();
    const iconNotification = shallow(<IconNotification icon={'notification'} notifications={5} onClickHandler={mockFn}/>);
    const iconNotificationMoreThanNinetynine = shallow(<IconNotification icon={'notification'} notifications={100} onClickHandler={mockFn}/>);

    it('should be defined', () => {
        expect(IconNotification).toBeDefined();
    });
    it('should render correctly', () => {
        expect(iconNotification).toMatchSnapshot();
        expect(iconNotificationMoreThanNinetynine).toMatchSnapshot();
    });
    it('IconNotification click event', () => {
        iconNotification.find(`.${styles.iconNotification}`).simulate('click');
        expect(mockFn).toHaveBeenCalledTimes(1);
    });
});
