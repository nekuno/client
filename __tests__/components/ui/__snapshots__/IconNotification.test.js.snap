// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test IconNotification component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <IconNotification
    icon="notification"
    notifications={5}
    onClickHandler={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="icon icon icon-notification"
        />,
        <div
          className="notificationsWrapper"
        >
          <div
            className="notifications"
          >
            5
          </div>
        </div>,
      ],
      "className": "iconNotification",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "icon icon icon-notification",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="notifications"
          >
            5
          </div>,
          "className": "notificationsWrapper",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 5,
            "className": "notifications",
          },
          "ref": null,
          "rendered": 5,
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="icon icon icon-notification"
          />,
          <div
            className="notificationsWrapper"
          >
            <div
              className="notifications"
            >
              5
            </div>
          </div>,
        ],
        "className": "iconNotification",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "icon icon icon-notification",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="notifications"
            >
              5
            </div>,
            "className": "notificationsWrapper",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 5,
              "className": "notifications",
            },
            "ref": null,
            "rendered": 5,
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Test IconNotification component should render correctly 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <IconNotification
    icon="notification"
    notifications={100}
    onClickHandler={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="icon icon icon-notification"
        />,
        <div
          className="notificationsWrapper"
        >
          <div
            className="notifications"
          >
            99
          </div>
        </div>,
      ],
      "className": "iconNotification",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "icon icon icon-notification",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="notifications"
          >
            99
          </div>,
          "className": "notificationsWrapper",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 99,
            "className": "notifications",
          },
          "ref": null,
          "rendered": 99,
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="icon icon icon-notification"
          />,
          <div
            className="notificationsWrapper"
          >
            <div
              className="notifications"
            >
              99
            </div>
          </div>,
        ],
        "className": "iconNotification",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "icon icon icon-notification",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="notifications"
            >
              99
            </div>,
            "className": "notificationsWrapper",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 99,
              "className": "notifications",
            },
            "ref": null,
            "rendered": 99,
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
